src/c1_foeppl_von_karman/foeppl_von_karman_curvable_bell_elements.h

```diff
@@ -0,0 +1,1300 @@
+    {
+      BrokenCopy::broken_assign("FoepplVonKarmanC1CurvableBellElement");
+    }
+
+    /// Function pointer to basis vectors function which sets  basis vectors
+    /// b1 and b2 (which are in general functions of x)
+    typedef void (*BasisVectorsFctPt)(const Vector<double>& x,
+                                      Vector<double>& b1,
+                                      Vector<double>& b2,
+                                      DenseMatrix<double>& Db1,
+                                      DenseMatrix<double>& Db2);
+
+    /// Get the zeta coordinate
+    inline void interpolated_zeta(const Vector<double>& s,
+                                  Vector<double>& zeta) const
+    {
+      // If there is a macro element use it
+      if (this->Macro_elem_pt != 0)
+      {
+        this->get_x_from_macro_element(s, zeta);
+      }
+      // Otherwise interpolate zeta_nodal using the shape functions
+      else
+      {
+        interpolated_x(s, zeta);
+      }
+    }
+
+    /// Lagrange interpolated shape for in--plane displacements
+    void shape_u(const Vector<double>& s, Shape& psi) const;
+
+    /// Lagrange interpolated d_shape for in--plane displacements
+    void dshape_u_local(const Vector<double>& s,
+                        Shape& psi,
+                        DShape& dpsids) const;
+
+    /// Add the element's contribution to its residual vector (wrapper) with
+    /// cached association matrix
+    void fill_in_contribution_to_residuals(Vector<double>& residuals)
+    {
+      // Store the expensive-to-construct matrix
+      this->store_association_matrix();
+      // Call the generic routine with the flag set to 1
+      FoepplVonKarmanEquations::fill_in_contribution_to_residuals(residuals);
+      // Remove the expensive-to-construct matrix
+      this->delete_association_matrix();
+    }
+
+    /// Add the element's contribution to its residual vector and
+    /// element Jacobian matrix (wrapper) with caching of association matrix
+    void fill_in_contribution_to_jacobian(Vector<double>& residuals,
+                                          DenseMatrix<double>& jacobian)
+    {
+      // Store the expensive-to-construct matrix
+      this->store_association_matrix();
+      // Call the generic routine with the flag set to 1
+      FoepplVonKarmanEquations::fill_in_contribution_to_jacobian(residuals,
+                                                                 jacobian);
+      // Remove the expensive-to-construct matrix
+      this->delete_association_matrix();
+    }
+
+    //----------------------------------------------------------------------------
+    // Interface to FoepplVonKarmanEquations (can this all be (static) data?)
+
+    /// Field indices for u
+    virtual Vector<unsigned> u_field_indices() const
+    {
+      return {0, 1};
+    }
+
+    /// Field indices for u
+    virtual unsigned u_alpha_field_index(const unsigned& alpha) const
```

Comment:
Do we really need the alpha. I think of this (and elsewhere) as the vector u, and the argument alpha makes it clear that we're asking for a single component. I feel the alpha in the function name is confusing.